### ISSUE TITLE
Since 4.1.19: xbmc.Playlist error when starting a TV channel

### ISSUE THREAD

**Issue by hoffleit** [MediaBrowser/plugin.video.emby#333]
The ChangeLog to 4.1.19 says "report back if you still get some play failures."

Well, since 4.1.19 (as far as I can remember), I get an error when I start playback of a TV channel: "Error Contents: an integer is required" within "self.playlist = xbmc.PlayList(self.monitor.playlistid)".

This error only happens when no Emby movie was played before during that Kodi session. To fix the error for that session, I have to start an Emby movie once.

This error also doesn't occur when I deactivate the Emby plugin. It's back as soon as a active Emby again.

This is the exception:

```
2019-12-19 22:37:52.412 T:3230167936  NOTICE: EMBY.objects.player -> [ onAVStarted ]
2019-12-19 22:37:52.413 T:3319788416  NOTICE: EMBY.hooks.monitor -> -->[ q:monitor/Player.OnAVChange ]
2019-12-19 22:37:52.416 T:3230167936   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.TypeError'>
                                            Error Contents: an integer is required
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/userdata/addon_data/plugin.video.emby/emby/objects/player.py", line 97, in onAVStarted
                                                item = self.set_item(current_file, False)
                                              File "/storage/.kodi/addons/plugin.video.emby/resources/lib/hooks/player.py", line 271, in set_item
                                                self.playlist = xbmc.PlayList(self.monitor.playlistid)
                                            TypeError: an integer is required
                                            -->End of Python script error report<--
2019-12-19 22:37:52.566 T:3169760128  NOTICE: EMBY.hooks.monitor -> -->[ q:monitor/Player.OnAVChange ]
```

I'm running CoreElec 9.2.1 with TVHeadend as PVR and Emby 4.1.19.

**Comment by angelblue05** [comment]
Try this build
http://kodi.emby.media/Public%20testing/plugin.video.emby-4.1.19a.zip

**Comment by hoffleit** [comment]
Sorry, I thought I already had replied to this.

The error when starting a TV channel is gone with 4.1.19a.

BUT: Kodi Sync doesn't work any more: I still see "EMBY.entrypoint.service -> [ LibraryChanged ]" notifications, but no "EMBY.library -> ---[ updated:1 ]" is happening with 4.1.19a. When I go back to 4.1.19, updates work again.

**Comment by angelblue05** [comment]
Could u send me ur kodi log via emby forums pm? Thanks!

**Comment by hoffleit** [comment]
@angelblue05, regarding the reported problem, I can confirm that this isolated change from 4.1.19a fixed makes the error message go away:

--- ./resources/lib/hooks/player.py.ORIG	2020-01-26 18:39:07.000000000 +0100
+++ ./resources/lib/hooks/player.py	2020-01-25 16:12:52.000000000 +0100
@@ -268,7 +268,7 @@
 
         ''' Call when playback start to setup play entry in player tracker.
         '''
-        self.playlist = xbmc.PlayList(self.monitor.playlistid)
+        self.playlist = xbmc.PlayList(self.monitor.playlistid or 1)
 
         if not file:
             LOG.warn("Filename is invalid")

There are other changes in 4.1.19a, Isee. Some of them break Kodi Emby Sync for me. I would have to prepare a kodi.log first, though, if you want to go into this.

Regarding this issue, the fix above seems to be enough.

**Comment by quickmic** [comment]
Will be fixed in next release

**Comment by quickmic** [comment]
Fixed in 4.1.22, Please check
